fix version

cgea1 opened the pull request.

cgea1 committed:
fix version

kissifrot commented:
Qu'est-ce qui ne fonctionnait pas du coup ?

kissifrot commented:
> Qu'est-ce qui ne fonctionnait pas du coup ?

Vu ensemble, le plugin ne démarrait pas :memo:

kissifrot committed:
Update plugin version to show changes

kissifrot committed:
Fix probable high memory usage by storing only useful parts of the JWT Token

kissifrot committed:
Improve string concatenation performance

kissifrot commented:
J'ai rajouté une amélioration de perfs :
```
   │ Lua version: Lua 5.1                                                                                                                                                             │
   │                                                                                                                                                                                  │
   │ ==============================================================                                                                                                                   │
   │ Test 1: Token JWT typique (10 roles, 5 groups)                                                                                                                                   │
   │ ==============================================================                                                                                                                   │
   │ Method                   Time (s)         Ops/sec     Mem (KB)                                                                                                                   │
   │ --------------------------------------------------------------                                                                                                                   │
   │ Old (concat)               0.5688           87900         0.00                                                                                                                   │
   │ New (table.concat)         0.4649          107548         0.00                                                                                                                   │
   │                                                                                                                                                                                  │
   │ => Amélioration: 1.22x plus rapide                                                                                                                                               │
   │                                                                                                                                                                                  │
   │ ==============================================================                                                                                                                   │
   │ Test 2: Token JWT moyen (50 roles, 20 groups)                                                                                                                                    │
   │ ==============================================================                                                                                                                   │
   │ Method                   Time (s)         Ops/sec     Mem (KB)                                                                                                                   │
   │ --------------------------------------------------------------                                                                                                                   │
   │ Old (concat)               0.3508           28507         0.00                                                                                                                   │
   │ New (table.concat)         0.1814           55133         0.00                                                                                                                   │
   │                                                                                                                                                                                  │
   │ => Amélioration: 1.93x plus rapide                                                                                                                                               │
   │                                                                                                                                                                                  │
   │ ==============================================================                                                                                                                   │
   │ Test 3: Token JWT volumineux (200 roles, 100 groups)                                                                                                                             │
   │ ==============================================================                                                                                                                   │
   │ Method                   Time (s)         Ops/sec     Mem (KB)                                                                                                                   │
   │ --------------------------------------------------------------                                                                                                                   │
   │ Old (concat)               0.1647            6072         0.00                                                                                                                   │
   │ New (table.concat)         0.0512           19516         0.00                                                                                                                   │
   │                                                                                                                                                                                  │
   │ => Amélioration: 3.21x plus rapide                                                                                                                                               │
   │                                                                                                                                                                                  │
   │ ==============================================================                                                                                                                   │
```

kissifrot committed:
Replace deprecated table.getn() by # syntax

 │ Lua version: Lua 5.1                                                                                                                                                             │
   │                                                                                                                                                                                  │
   │ ==================================================                                                                                                                               │
   │ Test: allowed_iss (2 éléments)                                                                                                                                                   │
   │ ==================================================                                                                                                                               │
   │ Method                   Time (s)         Ops/sec                                                                                                                                │
   │ --------------------------------------------------                                                                                                                               │
   │ table.getn()               0.0331        30183212                                                                                                                                │
   │ # operator                 0.0142        70506945                                                                                                                                │
   │                                                                                                                                                                                  │
   │ => Amélioration: 2.34x plus rapide                                                                                                                                               │
   │                                                                                                                                                                                  │
   │ ==================================================                                                                                                                               │
   │ Test: allowed_scopes (5 éléments)                                                                                                                                                │
   │ ==================================================                                                                                                                               │
   │ Method                   Time (s)         Ops/sec                                                                                                                                │
   │ --------------------------------------------------                                                                                                                               │
   │ table.getn()               0.0336        29776970                                                                                                                                │
   │ # operator                 0.0164        61094819                                                                                                                                │
   │                                                                                                                                                                                  │
   │ => Amélioration: 2.05x plus rapide                                                                                                                                               │
   │                                                                                                                                                                                  │
   │ ==================================================                                                                                                                               │
   │ Test: realm_roles (20 éléments)                                                                                                                                                  │
   │ ==================================================                                                                                                                               │
   │ Method                   Time (s)         Ops/sec                                                                                                                                │
   │ --------------------------------------------------                                                                                                                               │
   │ table.getn()               0.0340        29436006                                                                                                                                │
   │ # operator                 0.0182        55023660                                                                                                                                │
   │                                                                                                                                                                                  │
   │ => Amélioration: 1.87x plus rapide                                                                                                                                               │
   │                                                                                                                                                                                  │
   │ ==================================================                                                                                                                               │
   │ Test: client_roles (50 éléments)                                                                                                                                                 │
   │ ==================================================                                                                                                                               │
   │ Method                   Time (s)         Ops/sec                                                                                                                                │
   │ --------------------------------------------------                                                                                                                               │
   │ table.getn()               0.0362        27645693                                                                                                                                │
   │ # operator                 0.0204        49034030                                                                                                                                │
   │                                                                                                                                                                                  │
   │ => Amélioration: 1.77x plus rapide                                                                                                                                               │
   │                                                                                                                                                                                  │
   │ ==================================================                                                                                                                               │
   │ Test: Pattern 'if arr == nil or len == 0'                                                                                                                                        │
   │ ==================================================                                                                                                                               │
   │ Method                   Time (s)         Ops/sec                                                                                                                                │
   │ --------------------------------------------------                                                                                                                               │
   │ table.getn()               0.0382        26178696                                                                                                                                │
   │ # operator                 0.0200        50045041                                                                                                                                │
   │                                                                                                                                                                                  │
   │ => Amélioration: 1.91x plus rapide                                                                                                                                               │
   │

kissifrot committed:
Another performance improvement in validators (set lookup vs nested loops)

     Lua version: Lua 5.1

     Test 1: Realm roles validation (10 roles, 3 allowed)
       Old (nested loops):    0.0845s
       New (set lookup):      0.0496s
       => Amélioration: 1.70x plus rapide

     Test 2: Realm roles validation (50 roles, 6 allowed)
       Old (nested loops):    0.6729s
       New (set lookup):      0.1218s
       => Amélioration: 5.52x plus rapide

     Test 3: Realm roles validation (200 roles, 11 allowed)
       Old (nested loops):    4.1756s
       New (set lookup):      0.4097s
       => Amélioration: 10.19x plus rapide

     Test 4: Client roles validation (3 apps, 8 roles)
       Old (nested loops+gmatch): 0.0460s
       New (set lookup):          0.0371s
       => Amélioration: 1.24x plus rapide
